### PR TITLE
#19 add readme

### PR DESCRIPTION
**Pull request by Emily-Jiang**
Signed-off-by: Emily Jiang <emijiang@uk.ibm.com>

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -15,8 +15,63 @@
...
+
+image:https://badges.gitter.im/eclipse/microprofile-service-mesh.svg[link="https://gitter.im/eclipse/microprofile-service-mesh"]
+
+=== Rationale
+
+Based on some link:https://groups.google.com/forum/#!searchin/microprofile/istio%7Csort:date/microprofile/7obnAXjt3QA/k4htskrcBwAJ [discussion], we need to investigate the relationship between MicroProfile and Service Mesh. 
+
+Why? 
+
+MicroProfile defines programming model for developing cloud-native microservices. Cloud-native microservices normally run in a service mesh. It is important for MicroProfile to understand the capabilities of service mesh, so that MicroProfile can offer complimentary features for the infrastructure and avoid the conflicts. Let's find out what they can offer first.
```
Comment (pilhuhn) [inline README.adoc]: defines *a* programming model
I would not say that they 'normally' run in a service mesh, as the service mesh concept and especially Istio is still quite new
capability of *a* service mesh   or  capability of service mesh*es*

Comment (Emily-Jiang) [reply README.adoc]: how about:
Ideally, Cloud-native microservices are deployed in a service mesh.

Comment (kenfinnigan) [reply README.adoc]: I'd prefer something like: "Cloud Native microservices developed with MicroProfile can take advantage of a Service Mesh by extracting many concerns away from the development of the microservice itself"

File: README.adoc [hunk 2]
```diff
@@ -15,8 +15,63 @@
...
+There are two popular service mesh as follows:
+* link:"https://istio.io/"[Istio] 
+* link:"https://linkerd.io/"[Linkerd]
+The above two implementations are both open-source projects and designed for cloud-natvie microservices. Istio uses Lyfts' Envoy as a sidecar proxy while Linkerd is built on top of Netty and Finagle. 
+Istio offers more functionalities than Linkerd, such as enforcing access control and usage policies across the service mesh. Refer to this link:https://abhishek-tiwari.com/a-sidecar-for-your-service-mesh/[page] for more comparison
+
+This specification looks at the service mesh in general with more focus on Istio.
+
+=== What is MicroProfile?
+MicroProfile defines programming model for developing cloud-native microservices. It offers the following capabilities:
```
Comment (kenfinnigan) [inline README.adoc]: Do we really need to list out all the MP specs here?

Will be a pain to update

File: README.adoc (hunk 1 above)
Comment (Emily-Jiang) [reply README.adoc]: updated based on what @kenfinnigan suggested.

File: README.adoc (hunk 2 above)
Comment (Emily-Jiang) [reply README.adoc]: removed the list

**Comment by Emily-Jiang** [comment]
@pilhuhn @kenfinnigan pls re-review. I would like to merge this soon.

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**